integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/standalone/ratelimit/OperationLevelRatelimitTestcase.java

```diff
@@ -0,0 +1,81 @@
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.choreo.connect.tests.testcases.standalone.ratelimit;
+
+
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.choreo.connect.tests.common.model.API;
+import org.wso2.choreo.connect.tests.common.model.ApplicationDTO;
+import org.wso2.choreo.connect.tests.testcases.withapim.throttle.ThrottlingBaseTestCase;
+import org.wso2.choreo.connect.tests.util.ApictlUtils;
+import org.wso2.choreo.connect.tests.util.TestConstant;
+import org.wso2.choreo.connect.tests.util.TokenUtil;
+import org.wso2.choreo.connect.tests.util.Utils;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class OperationLevelRatelimitTestcase {
+    private String testKey;
+
+    @BeforeClass
+    public void createApiProject() throws Exception {
+        ApictlUtils.createProject("rateLimit_openAPI.yaml", "ratelimit_operation_level_test",
+                null, null, null, "operation_level_ratelimit_api.yaml");
+        ApictlUtils.login("test");
+        ApictlUtils.deployAPI("ratelimit_operation_level_test", "test");
+        Utils.delay(5000, "Could not wait till initial setup completion.");
+
+        API api = new API();
+        api.setName("ratelimit");
+        api.setContext("v2/ratelimitService");
+        api.setVersion("1.0.0");
+        api.setProvider("admin");
+
+        ApplicationDTO applicationDto = new ApplicationDTO();
+        applicationDto.setName("jwtApp");
+        applicationDto.setTier("Unlimited");
+        applicationDto.setId((int) (Math.random() * 1000));
+
+        testKey = TokenUtil.getJWT(api, applicationDto, "Unlimited", TestConstant.KEY_TYPE_PRODUCTION,
+                3600, null, true);
+    }
+
+    @Test(description = "Test operation level 3 permin rate-limiting with envoy rate-limit service")
+    public void testRateLimitsWithEnvoyRateLimitService3PerMin() throws Exception {
+        Map<String, String> headers = new HashMap<>();
+        headers.put("Internal-Key", testKey);
+        Utils.delay(3000, "Could not wait till initial setup completion.");
+        String endpointURL = Utils.getServiceURLHttps("/v2/ratelimitService/pet/findByStatus");
+        Assert.assertTrue(ThrottlingBaseTestCase.isThrottled(endpointURL, headers, null, 3),
+                "Operation level rate-limit 3 per min testcase failed.");
+    }
+
+    @Test(description = "Test operation level 5 per min rate-limiting with envoy rate-limit service")
+    public void testRateLimitsWithEnvoyRateLimitService5PerMin() throws Exception {
+        Map<String, String> headers = new HashMap<>();
+        headers.put("Internal-Key", testKey);
+        Utils.delay(3000, "Could not wait till initial setup completion.");
+        String endpointURL = Utils.getServiceURLHttps("/v2/ratelimitService/pet/3");
```

Comment:
Is there any possibility to test with different pet IDs? for example pet/2, pet/6